src/mmw/js/src/draw/views.js

```diff
@@ -194,6 +198,40 @@ var DrawAreaView = Marionette.ItemView.extend({
         }).always(function() {
             self.model.enableTools();
         });
+    },
+
+    enableStampTool: function() {
+        var self = this,
+            map = App.getLeafletMap(),
+            revertLayer = clearAoiLayer();
+
+        this.model.disableTools();
+        utils.placeMarker(map).then(function(latlng) {
+            var point = L.marker(latlng).toGeoJSON(),
+                halfKmbufferPoints = _.map([-180, -90, 0, 90], function(bearing) {
+                    var p = turfDestination(point, 0.5, bearing, 'kilometers');
+                    return L.latLng(p.geometry.coordinates[1], p.geometry.coordinates[0]);
+                }),
+                // Convert the four points into two SW and NE for the bounding
+                // box. Do this by splitting the array into two arrays of two
+                // points. Then map each array of two to a single point by
+                // taking the lat from one and lng from the other.
+                swNe = _.map(_.toArray(_.groupBy(halfKmbufferPoints, function(p, i) {
+                    // split the array of four in half.
+                    return i < 2;
+                })), function(pointGroup) {
+                    return L.latLng(pointGroup[0].lat, pointGroup[1].lng);
+                }),
+                bounds = L.latLngBounds(swNe),
+                box = turfBboxPolygon(bounds.toBBoxString().split(','));
+
+            addLayer(box, '1 Square Km');
+            navigateToAnalyze();
+        }).fail(function() {
```

Comment:
Should this `fail` execute when `placeMarker` fails, or when `function(latlng)` fails? Since we're using `.then` instead of `.done`, currently it will execute in the latter case, but I'm not sure if it will in the former.